redundant-bmc/src/sibling_impl.cpp

```diff
@@ -0,0 +1,302 @@
+
+                if (!serviceName.empty())
+                {
+                    ctx.spawn(watchNameOwnerChanged());
+                }
+            }
+        }
+    }
+    co_return;
+}
+
+// NOLINTNEXTLINE
+sdbusplus::async::task<> SiblingImpl::watchInterfaceRemoved()
+{
+    namespace rules = sdbusplus::bus::match::rules;
+    sdbusplus::async::match match(ctx,
+                                  rules::interfacesRemovedAtPath(objectPath));
+
+    while (!ctx.stop_requested())
+    {
+        auto [_, interfaces] =
+            co_await match.next<sdbusplus::message::object_path,
+                                std::vector<std::string>>();
+
+        if (std::ranges::contains(interfaces,
+                                  redundancy_ns::Sibling::interface))
+        {
+            lg2::info("Sibling D-Bus interface removed");
+            interfacePresent = false;
+            heartbeat = false;
+        }
+    }
+
+    co_return;
+}
+
+// NOLINTNEXTLINE
+sdbusplus::async::task<> SiblingImpl::watchPropertyChanged()
+{
+    namespace rules = sdbusplus::bus::match::rules;
+    sdbusplus::async::match match(
+        ctx, rules::propertiesChanged(objectPath,
+                                      redundancy_ns::Sibling::interface));
+
+    while (!ctx.stop_requested())
+    {
+        auto [iface,
+              propertyMap] = co_await match.next<std::string, PropertyMap>();
+
+        for (const auto& [name, value] : propertyMap)
+        {
+            lg2::info("Sibling property {PROP} changed", "PROP", name);
+        }
+
+        loadFromPropertyMap(propertyMap);
+    }
+
+    co_return;
+}
+
+// NOLINTNEXTLINE
+sdbusplus::async::task<> SiblingImpl::watchNameOwnerChanged()
+{
+    namespace rules = sdbusplus::bus::match::rules;
+    sdbusplus::async::match match(ctx, rules::nameOwnerChanged(serviceName));
+
+    while (!ctx.stop_requested())
+    {
+        auto [name, oldOwner, newOwner] =
+            co_await match.next<std::string, std::string, std::string>();
+
+        if (!oldOwner.empty() && newOwner.empty())
+        {
+            lg2::info("Sibling D-Bus name lost");
+            interfacePresent = false;
+            heartbeat = false;
+        }
+    }
+    co_return;
+}
+
+// NOLINTBEGIN
+sdbusplus::async::task<>
+    SiblingImpl::waitForSiblingUp(const std::chrono::seconds& timeout)
+// NOLINTEND
+{
+    using namespace std::chrono_literals;
+    auto start = std::chrono::steady_clock::now();
+    auto waiting = false;
+
+    while ((!interfacePresent || !heartbeat) &&
+           ((std::chrono::steady_clock::now() - start) < timeout))
+    {
+        if (!waiting)
+        {
+            lg2::info("Waiting for sibling interface and/or heartbeat: "
+                      "Present = {PRES}, Heartbeat = {HB}",
+                      "PRES", interfacePresent, "HB", heartbeat);
+            waiting = true;
+        }
+
+        co_await sdbusplus::async::sleep_for(ctx, 500ms);
+    }
+
+    lg2::info(
```

Comment:
Yea.  I'm going to handle enabling redundancy kinda like determining the role, where it'll pass in all of  the relevant parameters to a function that will spit out if redundancy can be enabled and the reasons why it can't be if not.